paasta_tools/check_flink_services_health.py

```diff
@@ -53,6 +57,66 @@ def healthy_flink_containers_cnt(si_pods: Sequence[V1Pod], container_type: str)
     )
 
 
+def send_event_if_not_enough_taskmanagers(
+    instance_config: FlinkDeploymentConfig,
+    expected_count: int,
+    num_reported: int,
+    strerror: str,
+) -> None:
+    under_replicated = False
+    if strerror is None:
+        crit_threshold = instance_config.get_replication_crit_percentage()
+        output = (
+            "Service %s has %d out of %d expected instances of %s reported by dashboard!\n"
+            + "(threshold: %d%%)"
+        ) % (
+            instance_config.job_id,
+            num_reported,
+            expected_count,
+            "taskmanager",
+            crit_threshold,
+        )
+        under_replicated, _ = is_under_replicated(
+            num_reported, expected_count, crit_threshold
+        )
+    else:
+        output = ("Dashboard of service %s is not available!\n" + "(%s)") % (
+            instance_config.job_id,
+            strerror,
+        )
+    if under_replicated or strerror:
+        output += (
+            "\n\n"
+            "What this alert means:\n"
+            "\n"
+            "  This alert means that the Flink service is not reporting the\n"
+            "  requested number of taskmanagers.\n"
+            "\n"
+            "Reasons this might be happening:\n"
+            "\n"
+            "  The service may simply be unhealthy. There also may not be enough resources\n"
+            "  in the cluster to support the requested instance count.\n"
+            "\n"
+            "Things you can do:\n"
+            "\n"
+            "  * Fix the cause of the unhealthy service. Try running:\n"
+            "\n"
+            "      paasta status -s %(service)s -i %(instance)s -c %(cluster)s -vv\n"
```

Comment:
this is probably copy-pasted from somewhere, but a multiline string would be much more readable here.